bundle/LegacyMapper/Configuration.php

```diff
@@ -180,6 +180,9 @@ public function onBuildKernel( PreBuildKernelEvent $event )
         // Multisite settings (PathPrefix and co)
         $settings += $this->getMultiSiteSettings();
 
+        // SiteAccess settings
+        $settings += $this->getSiteSettings();
```

Comment:
My point is, it is to complex, it is hard to read.

maybe *remove* `getSiteSettings`, and move the logic into the `if ( $rootLocationId === null )` block, with inline doc saying we return plan site settings if rootLocationId is not used.